crates/adapters/dydx/src/websocket/client.rs

```diff
@@ -72,6 +78,14 @@ pub struct DydxWebSocketClient {
     instruments_cache: Arc<DashMap<Ustr, InstrumentAny>>,
     /// Optional account ID for account message parsing.
     account_id: Option<AccountId>,
+    /// Optional heartbeat interval in seconds.
+    heartbeat: Option<u64>,
+    /// Inner websocket client holder (with reconnection and handler mode).
+    inner: Arc<RwLock<Option<WebSocketClient>>>,
```

Comment:
There's been some architectural changes: handler now owns the base websocket inside the lock-free I/O network boundary, removes the `RwLock` contention (see `adapters.md`).

Reply:
ok, u already addressed this here

https://github.com/nautechsystems/nautilus_trader/commit/77de0be507207c021b8003c04dfacaae7ff2efb8?diff=unified


refactored this in the current PR as well